core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -65,6 +63,16 @@ object Json {
     catch { case e: JsonProcessingException => Left(e) }
   }
 
+  /**
+    * Parse a JSON string into a JsonValue if possible. It returns an `Either` where `Left` will be an exception and
+    * `Right` is the `JsonValue`.
+    * @param input a JSON string to parse
+    * @return An `Either` which in case of `Left` means an exception and `Right` is the actual return value.
+    */
+  def tryParseFull(input: String): Either[JsonProcessingException, JsonValue] =
+    try Right(mapper.readTree(input)).map(JsonValue(_))
```

Comment:
Checking Jackson's documentation, it says:

> If no content is found from input (end-of-input), Java
>      * <code>null</code> will be returned.

So, we probably need to handle this ourselves and convert the returned `null` to an exception. The same issue affects other methods in this class that return `Either`.

Reply:
I don't think that's possible in this case as with the String parameter you pass in the Json string itself which can either be `null` in which case an `IllegalArgumentException` will be thrown or some kind of value which can or can't be parsed and then the method returns with `Either`.
In the cases where you read the Json in from an InputStream or a Reader it is possible to return `null` as you can terminate the stream without providing any actual content (but in the String you provide the content itself).
If you want, I can handle the possible `null` but I don't think it's possible with `readTree(String)`.

Reply:
Can we just add a test for `null` String and `null` `Array[Byte]` (the other overload) if we don't have them? If they pass, we're good.

Reply:
Added them